02-IDE_exercises/solutions/07b-project/reverse_list.py

```diff
@@ -1,14 +1,22 @@
-# Author: Brandon Hoffman
-# Date: 10/29/2020
-# Description: Mutates a given list and by the order of the elements of teh list
+#Author: Luis Mercado
+#Date: 10/24/2024
+#Description: Function that mutates a list of numbers by reversing its order without slicing
 
-def reverse_list(lst):
+
+def reverse_list(num_list: list[int | float]) -> None:
     """
-    Input: list
-    Takes in a list and reverses the order of the elements of the list
+    Mutates list of numbers (float or int) into reverse order
+    without using any slicing (no return value)
     """
-    temp_lst = []
-    for i in range(len(lst)-1,-1, -1):
-        temp_lst.append(lst[i])
     
-    lst[:] = temp_lst
\ No newline at end of file
+    #define lower limit of desired index range which is the same as upper limit of starting list argument
+    low_lim: int = len(num_list)-1
+    
+    #for range must use negative stride and go to -1 so that 0 index is included
+    for num in range(low_lim, -1, -1):
+        num_list.append(num_list[num]) #adds currently indexed list value to end of list
+        del num_list[num] #deletes original iteration of currently indexed value from list
+
+    #no return value, but by using .append and del[], the original num_list argument has iteratively had its last
```

Comment:
I prefer there to be a space after a comment. Looks better to my eyes. Can you add spaces after the `#`?

Reply:
Agreed it looks better! Updated all comments in my local clone and committed those changes to this branch (I can see it as a separate commit within my pull request) -- is there anything else I need to do before the changes can be merged / will the merge happen correctly?